ObjectiveGit/GTBranch.m

```diff
@@ -193,7 +193,12 @@ - (GTBranch *)trackingBranchWithError:(NSError **)error success:(BOOL *)success
 }
 
 - (BOOL)updateTrackingBranch:(GTBranch *)trackingBranch error:(NSError **)error {
-	int result = git_branch_set_upstream(self.reference.git_reference, trackingBranch.shortName.UTF8String);
+  int result = GIT_ENOTFOUND;
+  if (trackingBranch.branchType == GTBranchTypeRemote) {
+    result = git_branch_set_upstream(self.reference.git_reference, [trackingBranch.name stringByReplacingOccurrencesOfString:[GTBranch remoteNamePrefix] withString:@""].UTF8String);
```

Comment:
Oh sorry, I misunderstood the problem!

I guess the opposite question, then: does it make sense to keep the current, buggy case? Should we just support setting the upstream to a remote branch?